examples/with-i18n-next-intl/pages/_app.js

```diff
@@ -0,0 +1,35 @@
+import { NextIntlProvider } from 'next-intl'
+import NextApp from 'next/app'
+
+export default function App({ Component, messages, pageProps }) {
+  // You can merge messages that should always be present
+  // (from `App.getInitialProps`) with page-level
+  // messages (from `getStaticProps` of individual pages)
+  const allMessages = { ...messages, ...pageProps.messages }
+
+  // To achieve consistent date, time and number formatting
+  // across the app, you can define a set of global formats.
+  const formats = {
+    dateTime: {
+      short: {
+        day: 'numeric',
+        month: 'short',
+        year: 'numeric',
+      },
+    },
+  }
+
+  return (
+    <NextIntlProvider formats={formats} messages={allMessages}>
+      <Component {...pageProps} />
+    </NextIntlProvider>
+  )
+}
+
+App.getInitialProps = async function getInitialProps(context) {
```

Comment:
This is opting pages out of automatic static optimization. Could we not include this? We want to help uses fall into the pit of success by having pre-rendered pages by default if there's non-blocking data requirements.

Reply:
I've adapted the loading of translations now to work equally well for the pre-rendering feature of Next.js now 🙂 .